src/abstract-wallet.ts

```diff
@@ -650,6 +650,15 @@ export abstract class AbstractWallet {
         token: TokenLike,
         erc20ApproveThreshold: BigNumber = ERC20_APPROVE_TRESHOLD
     ): Promise<boolean> {
+
+        // validates token
+        try {
+            this.provider.tokenSet.resolveTokenSymbol(token)
+        } catch (error) {
+            throw error;
+        }
+        
+
```

Comment:
I would at least wrap it in a function that is called "validate".

Reply:
Removing the try-catch since that's a bit  redundant. Given this, asides from inherent readability, I don't see any other reason to wrap it again as it's a one liner. I also opted to use that method since it was already available in the class and doesn't have any side effects. Not incredibly pretty but it's decent enough not to warrant any major refactor